### PR TITLE
Fixes #10911 - reinitialize select2 on templated forms

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by domcleal** [comment]
@jlsherrill you appear to be testing the linked PR, could you confirm if this fixes the issue?  I don't think we have a working test case in Foreman core for me to try.

File: app/assets/javascripts/lookup_keys.js [hunk 1]
```diff
@@ -79,6 +80,7 @@ function add_child_node(item) {
     $(item).closest("form").trigger({type: 'nested:fieldAdded', field: field});
     $('a[rel="popover"]').popover({html: true});
     $('a[rel="twipsy"]').tooltip();
+    $(field).find('select').select2();
```
Comment (domcleal) [inline app/assets/javascripts/lookup_keys.js]: per 1b67311, please add allowClear here

**Comment by ares** [comment]
Added allowClear

**Comment by jlsherrill** [comment]
@domcleal tested and solved the issue i was seeing.

**Comment by dLobatog** [comment]
Merged as 030cb35245a01b59e44a8db2803290e21a41da7f, thanks @ares!

**Pull request closed without merging**